test/boost/error_injection_test.cc

```diff
@@ -40,6 +41,14 @@ SEASTAR_TEST_CASE(test_inject_noop) {
 
     auto f = errinj.inject("noop2", sleep_msec);
     BOOST_REQUIRE(f.available() && !f.failed());
+
+    errinj.enable("noop3");
+    f = errinj.inject_with_handler("noop3", [] (auto& handler) {
+        return handler.wait_for_message(db::timeout_clock::time_point::max());
+    });
+
+    BOOST_REQUIRE(f.available() && !f.failed());
```

Comment:
I don't get it - shouldn't the `wait_for_message` hang because you didn't send any messages? So `f` should not actually be available? This looks like a bug.

Reply:
Because this test case tests disabled injections (release mode). All `inject` and `inject_with_handler` should be noop.